### ISSUE TITLE
Doesn't write any file

### ISSUE THREAD

**Issue by fahrsoft** [VladimirKalachikhin/naiveGPXlogger#1]
I have tried various combinations of paths, including one in /home/pi/Documents/ and this one in the root, with different settings for permissions as either pi or root as owner and ability for all users to write to the folder, as well as ending with a slash or without. I've yet to have this plugin create a file after many days of having it enabled. Is there something wrong with the settings or am I missing something else? Thanks!!!

`{
  "configuration": {
    "logging": true,
    "trackFrequency": 120,
    "minmove": 0.1,
    "trackTimeout": 3600,
    "metadata": {
      "skipperName": false
    },
    "depthProp": {
      "enable": false,
      "feature": "Depth below transducer (DBT)",
      "fixDepth": false
    },
    "trackDir": "/gpxlogs",
    "everyDay": true
  },
  "enabled": true,
  "enableDebug": true,
  "enableLogging": true
}`

Here's a snip of the server log after changing the path:
`
Jan 19 15:46:55 Front pattern: FFR 
Jan 19 15:46:55 Handle: {"path":"environment.outside.pressure","value":101776.85104697186} 
Jan 19 15:46:59 Handle: {"path":"navigation.position","value":{"longitude":-81.02207233333333,"latitude":29.232968333333332}} 
Jan 19 15:47:01 Handle: {"path":"navigation.gnss.antennaAltitude","value":-0.1} 
Jan 19 15:47:17 GET /skServer/plugins 304 32.699 ms - - 
Jan 19 15:47:24 2023-01-19T20:47:24.056Z naivegpxlogger plugin started, now logging is true log dir is /gpxlogs/ 
Jan 19 15:47:24 POST /skServer/plugins/naivegpxlogger/config 200 9.702 ms - 45 
Jan 19 15:47:24 GET /skServer/plugins 200 23.968 ms - - 
Jan 19 15:47:26 GET /skServer/debugKeys 304 2.067 ms - - 
`

**Comment by VladimirKalachikhin** [comment]
It should work...
I see that
`naivegpxlogger plugin started, now logging is true log dir is /gpxlogs/`
At least in `/gpxlogs/`must be a .gpx file with gpx headings.
Is it not there?

**Comment by fahrsoft** [comment]
No... nothing gets created.

**Comment by VladimirKalachikhin** [comment]
Ok.
Make sure that logging is enabled.

In SignalK web admin panel go to 
Server > Plugin Config > naiveGPXlogger
check Write log and press Submit:
![Снимок экрана в 2023-01-20 10-44-12](https://user-images.githubusercontent.com/32364744/213644372-64691243-395c-4207-8a34-de4037d342e4.png)
![Снимок экрана в 2023-01-20 10-44-34](https://user-images.githubusercontent.com/32364744/213644423-521ca66d-8d95-4fea-8ab3-8c2fd57ea976.png)
In SignalK log should be something like:
```
  naivegpxlogger plugin stopped, logging= false +1m
  naivegpxlogger plugin started, now logging is true log dir is /home/stager/projects/webNav/signalk/map/public/track/ +2ms
  naivegpxlogger Log enabled, log file /home/stager/projects/webNav/signalk/map/public/track/2023-01-20T07:44:39.851Z.gpx +34ms
  naivegpxlogger Options saved by Logging switch +2ms
```
Open in browser (your SignalK address instead localhost ): `http://localhost:3000/signalk/v1/api/vessels/self`:
![Снимок экрана в 2023-01-20 10-45-30](https://user-images.githubusercontent.com/32364744/213645452-8c9287d0-eb52-471b-a6a6-8ddd6e6ac091.png)

Has the log recording turned on?

**Comment by fahrsoft** [comment]
From the server log:
Jan 22 09:16:03 2023-01-22T14:16:03.830Z naivegpxlogger plugin stopped, logging= false 
Jan 22 09:16:03 2023-01-22T14:16:03.834Z naivegpxlogger plugin started, now logging is true log dir is /gpxlogs/ 
Jan 22 09:16:03 POST /skServer/plugins/naivegpxlogger/config 200 4.872 ms - 45 
Jan 22 09:16:03 GET /skServer/plugins 304 17.103 ms - - 
Jan 22 09:16:18 Handle: {"path":"navigation.position","value":{"longitude":-80.54626966666666,"latitude":27.990538666666666}} 
Jan 22 09:16:18 Front pattern: FRR 
Jan 22 09:16:18 Handle: {"path":"environment.outside.pressure","value":101729.01848381972} 
Jan 22 09:16:18 Handle: {"path":"environment.outside.temperature","value":300.59} 

From the SignalK API:
"navigation":{"trip":{"logging":{"meta":{},"value":{"status":false,"logFile":null},"$source":"naivegpxlogger.XX","timestamp":"2023-01-22T14:16:03.860Z"},"log":{"meta":{"units":"m","description":"Total distance traveled on this trip / since trip reset"},"value":3069783,"$source":"n2k.35","timestamp":"2023-01-22T14:17:55.101Z","pgn":128275}}
I’ve tried changing the values and the depth recording but no luck.

**Comment by fahrsoft** [comment]
![Screen Shot 2023-01-22 at 9 24 22 AM](https://user-images.githubusercontent.com/48223424/213921249-8bff315f-7212-4017-85e2-8c20a26e3cc2.png)

**Comment by VladimirKalachikhin** [comment]
Is in the Connection & Plugin Status sections of SignalK web admin panel  any messages from **naivegpxlogger** plugin?
For example:
![Снимок экрана в 2023-01-22 17-41-47](https://user-images.githubusercontent.com/32364744/213921777-62351534-5ad4-4c35-8544-b2bb539b814a.png)

**Comment by fahrsoft** [comment]
No errors... shows that it should be writing. 
![Screen Shot 2023-01-22 at 9 52 54 AM](https://user-images.githubusercontent.com/48223424/213922354-526d7879-d19a-4d88-a12e-e62f2b41bbcd.png)

**Comment by VladimirKalachikhin** [comment]
Rename this file to index.js and place it to ~/.signalk/node_modules/naivegpxlogger/
Restart signalk-server
See `naivegpxlogger [openTrack] routeSaveName after update path=` in SignalK log
Is this path correct?

[index.js.TXT](https://github.com/VladimirKalachikhin/naiveGPXlogger/files/10474725/index.js.TXT)

**Comment by fahrsoft** [comment]
I don't see any change. The error is for influx, not the logger. The path that it's referencing here is correct. 

`Jan 22 10:54:17 Handle: {"path":"environment.outside.pressure","value":101671.7234870993} 
Jan 22 10:54:18 Handle: {"path":"navigation.position","value":{"longitude":-80.54636233333333,"latitude":27.990507166666667}} 
Jan 22 10:54:19 Handle: {"path":"navigation.gnss.antennaAltitude","value":-3.9} 
Jan 22 10:54:20 2023-01-22T15:54:20.281Z naivegpxlogger plugin stopped, logging= false 
Jan 22 10:54:20 2023-01-22T15:54:20.284Z naivegpxlogger plugin started, now logging is true log dir is /gpxlogs/ 
Jan 22 10:54:20 POST /skServer/plugins/naivegpxlogger/config 200 5.643 ms - 45 
Jan 22 10:54:20 GET /skServer/plugins 304 16.562 ms - - 
Jan 22 10:54:27 invalid input to toNmeaDegreesLatitude: null 
Jan 22 10:54:27 invalid input to toNmeaDegreesLatitude: null 
Jan 22 10:54:27 invalid input to toNmeaDegreesLatitude: null 
Jan 22 10:54:27 invalid input to toNmeaDegreesLatitude: null 
Jan 22 10:54:27 invalid input to toNmeaDegreesLatitude: null 
Jan 22 10:54:27 Error: invalid float value for field 'lat': null at ge.floatField (/home/pi/.signalk/node_modules/@influxdata/influxdb-client/dist/index.js:4:3366) at SKInflux.handleValue (/home/pi/.signalk/node_modules/signalk-to-influxdb2/dist/influx.js:80:19) at /home/pi/.signalk/node_modules/signalk-to-influxdb2/dist/plugin.js:36:86 at Array.forEach (<anonymous>) at /home/pi/.signalk/node_modules/signalk-to-influxdb2/dist/plugin.js:36:40 at Array.forEach (<anonymous>) at /home/pi/.signalk/node_modules/signalk-to-influxdb2/dist/plugin.js:35:39 at Array.forEach (<anonymous>) at FullSignalK.<anonymous> (/home/pi/.signalk/node_modules/signalk-to-influxdb2/dist/plugin.js:34:35) at FullSignalK.emit (node:events:525:35) at FullSignalK.addDelta (/usr/lib/node_modules/signalk-server/node_modules/@signalk/signalk-schema/dist/fullsignalk.js:52:8) at DeltaChain.doProcess (/usr/lib/node_modules/signalk-server/lib/deltachain.js:14:18) at DeltaChain.process (/usr/lib/node_modules/signalk-server/lib/deltachain.js:10:21) at Function.Server.app.handleMessage (/usr/lib/node_modules/signalk-server/lib/index.js:180:32) at Simple.<anonymous> (/usr/lib/node_modules/signalk-server/lib/pipedproviders.js:59:17) at Simple.emit (node:events:525:35) at addChunk (node:internal/streams/readable:315:12) at readableAddChunk (node:internal/streams/readable:289:9) at Simple.Readable.push (node:internal/streams/readable:228:10) at Simple._transform (/usr/lib/node_modules/signalk-server/node_modules/@signalk/streams/simple.js:109:8) at Simple.Transform._write (node:internal/streams/transform:205:23) at writeOrBuffer (node:internal/streams/writable:391:12) at _write (node:internal/streams/writable:332:10) at Simple.Writable.write (node:internal/streams/writable:336:10) at Nmea0183ToSignalK.ondata (node:internal/streams/readable:754:22) at Nmea0183ToSignalK.emit (node:events:513:28) at addChunk (node:internal/streams/readable:315:12) at readableAddChunk (node:internal/streams/readable:289:9) 
Jan 22 10:54:27 invalid input to toNmeaDegreesLatitude: null `

**Comment by VladimirKalachikhin** [comment]
It's strange...
Run with this file, and see naivegpxlogger messages. Must be some as:
```
  naivegpxlogger plugin started, now logging is true log dir is /net/ +0ms
  naivegpxlogger Recording is not enabled yet, value.logFile= null options.trackDir= /net/ +14ms
  naivegpxlogger Log enabled, log file /net/2023-01-22T15:19:02.913Z.gpx +17ms
  naivegpxlogger Options saved by Logging switch +1ms
```

[index.js.TXT](https://github.com/VladimirKalachikhin/naiveGPXlogger/files/10474914/index.js.TXT)

**Comment by fahrsoft** [comment]
Out sailing now. Will check tonight/tomorrow. Thanks for the quick responses!!!

**Comment by VladimirKalachikhin** [comment]
Run this. Should work.

[index.js.TXT](https://github.com/VladimirKalachikhin/naiveGPXlogger/files/10477786/index.js.TXT)

**Comment by fahrsoft** [comment]
Still nothing. 
I don't know if it matters, but I have multiple sources for position, and they are set up to be prioritized. I've done a full restart of the Pi and have a fix using the GPS that's connected to the Pi (rather than any of the items over the N2K network). I'm not really moving at the moment, but have set all the settings to zero for frequency and move distance and a timeout of 15. 

`"navigation":{"trip":{"logging":{"meta":{},"value":{"status":false,"logFile":null},"$source":"naivegpxlogger.XX","timestamp":"2023-01-23T14:09:13.539Z"}`

`
Jan 23 09:09:00 GET /signalk/v1/api/vessels/self 200 3.455 ms - - 
Jan 23 09:09:02 Handle: {"path":"navigation.position","value":{"longitude":-80.37313616666667,"latitude":27.663665166666668}} 
Jan 23 09:09:08 Handle: {"path":"navigation.gnss.antennaAltitude","value":-2.5} 
Jan 23 09:09:13 2023-01-23T14:09:13.530Z naivegpxlogger plugin stopped, logging= false 
Jan 23 09:09:13 2023-01-23T14:09:13.533Z naivegpxlogger plugin started, now logging is true log dir is /gpxlogs/ 
Jan 23 09:09:13 POST /skServer/plugins/naivegpxlogger/config 200 5.787 ms - 45 
Jan 23 09:09:13 GET /skServer/plugins 200 29.647 ms - - `

**Comment by VladimirKalachikhin** [comment]
That is, there are no other messages from naivegpxlogger after
```
Jan 23 09:09:13 2023-01-23T14:09:13.530Z naivegpxlogger plugin stopped, logging= false  
Jan 23 09:09:13 2023-01-23T14:09:13.533Z naivegpxlogger plugin started, now logging is true log dir is /gpxlogs/ 
```
?

**Comment by fahrsoft** [comment]
None.

**Comment by VladimirKalachikhin** [comment]
Try it:
[index.js.TXT](https://github.com/VladimirKalachikhin/naiveGPXlogger/files/10482238/index.js.TXT)

**Comment by fahrsoft** [comment]
Same thing... 
`"navigation":{"trip":{"logging":{"meta":{},"value":{"status":true,"logFile":null},"$source":"naivegpxlogger.XX","timestamp":"2023-01-23T22:31:15.953Z"}`

`

Jan 23 17:31:15 2023-01-23T22:31:15.944Z naivegpxlogger plugin started, now logging is true log dir is /gpxlogs/ 
Jan 23 17:31:15 2023-01-23T22:31:15.985Z naivegpxlogger Do establish navigation.trip.logging to true 
Jan 23 17:31:18 signalk-server running at 0.0.0.0:[object Object] 
Jan 23 17:31:19 Found BMx280 chip ID 0x60 on bus i2c-1, address 0x77 
Jan 23 17:31:19 Handle: {"path":"navigation.position","value":{"longitude":-80.37324033333333,"latitude":27.663565166666668}} 
Jan 23 17:31:19 Handle: {"path":"navigation.gnss.antennaAltitude","value":-17.3} 
Jan 23 17:31:19 BME280 initialization succeeded 
Jan 23 17:31:19 GET /admin/ 200 39.283 ms - 681 
Jan 23 17:31:19 Handle: {"path":"environment.outside.temperature","value":299.96999999999997} 
Jan 23 17:31:19 23 Jan 17:31:19 - [info] Welcome to Node-RED =================== 
Jan 23 17:31:19 23 Jan 17:31:19 - [info] Node-RED version: v3.0.2 
Jan 23 17:31:19 23 Jan 17:31:19 - [info] Node.js version: v16.19.0 
Jan 23 17:31:19 23 Jan 17:31:19 - [info] Linux 5.15.84-v8+ arm64 LE 
Jan 23 17:31:19 GET /@signalk/calibration/remoteEntry.js 304 79.799 ms - - 
Jan 23 17:31:19 GET /admin/main.js 304 5.631 ms - - 
Jan 23 17:31:19 GET /@signalk/vesselpositions/remoteEntry.js 304 29.576 ms - - 
Jan 23 17:31:19 GET /signalk-barograph/remoteEntry.js 304 15.803 ms - - 
Jan 23 17:31:19 GET /admin/784.js 304 7.931 ms - - 
Jan 23 17:31:19 GET /admin/316.js 304 7.058 ms - - 
Jan 23 17:31:19 GET /admin/436.js 304 6.207 ms - - 
Jan 23 17:31:19 GET /admin/270.js 304 4.432 ms - - 
Jan 23 17:31:19 GET /skServer/plugins 200 31.391 ms - - 
Jan 23 17:31:19 GET /skServer/addons 200 34.145 ms - 2 
Jan 23 17:31:20 GET /skServer/loginStatus 200 2.812 ms - 201 
Jan 23 17:31:20 GET /skServer/webapps 200 8.904 ms - - 
Jan 23 17:31:20 GET /signalk 200 8.043 ms - 267 
Jan 23 17:31:20 GET /skServer/security/access/requests 200 4.525 ms - 2 
Jan 23 17:31:20 GET /admin/fonts/signal-k-logo-image-text.d132ec2c4f0269c103c868aaa6868f01.svg 304 18.333 ms - - 
Jan 23 17:31:20 GET /admin/fonts/Simple-Line-Icons.3826fa1cb2348dd93948a50cbd2b8fb6.woff2 304 7.861 ms - - 
Jan 23 17:31:20 GET /admin/fonts/fa-solid-900.1551f4f60c37af51121f106501f69b80.woff2 304 8.670 ms - - 
Jan 23 17:31:20 GET /admin/fonts/fa-regular-400.7a3337626410ca2f40718481c755640f.woff2 304 5.051 ms - - 
Jan 23 17:31:20 23 Jan 17:31:20 - [info] Loading palette nodes 
Jan 23 17:31:22 23 Jan 17:31:22 - [info] Dashboard version 3.2.0 started at /plugins/signalk-node-red/redApi/ui 
Jan 23 17:31:22 23 Jan 17:31:22 - [info] Context store : 'default' [module=memory] 
Jan 23 17:31:22 23 Jan 17:31:22 - [info] User directory : /home/pi/.signalk/red 
Jan 23 17:31:22 23 Jan 17:31:22 - [warn] Projects disabled : set editorTheme.projects.enabled=true to enable 
Jan 23 17:31:22 23 Jan 17:31:22 - [warn] Flows file name not set. Generating name using hostname. 
Jan 23 17:31:22 23 Jan 17:31:22 - [info] Flows file : /home/pi/.signalk/red/flows_openplotter.json 
Jan 23 17:31:22 2023-01-23T22:31:22.517Z naivegpxlogger Subscribed to navigation.trip.logging 
Jan 23 17:31:22 23 Jan 17:31:22 - [info] Starting flows 
Jan 23 17:31:22 23 Jan 17:31:22 - [error] [e-mail in:get email] No e-mail userid set 
Jan 23 17:31:22 23 Jan 17:31:22 - [error] [e-mail in:get email] No e-mail password set 
Jan 23 17:31:23 23 Jan 17:31:23 - [info] Started flows 
Jan 23 17:31:23 23 Jan 17:31:23 - [info] [udp out:sendSK] udp ready: localhost:55559 
Jan 23 17:31:23 23 Jan 17:31:23 - [info] [mqtt-broker:4d32915d.41a748] Connected to broker: mqtt://localhost:1883 
Jan 23 17:31:24 GET /skServer/appstore/available 200 4165.416 ms - - 
Jan 23 17:31:28 GET /signalk/v1/api/vessels/self 200 13.275 ms - - 
Jan 23 17:32:19 Front pattern: FFS 
Jan 23 17:32:19 Handle: {"path":"environment.outside.pressure","value":102163.63328399608} 
Jan 23 17:32:20 Handle: {"path":"navigation.position","value":{"longitude":-80.37355783333334,"latitude":27.663551}} 
Jan 23 17:32:20 Handle: {"path":"navigation.gnss.antennaAltitude","value":-62.4} 
Jan 23 17:33:06 GET /skServer/debugKeys 200 4.205 ms - - 
Jan 23 17:33:17 Handle: {"path":"environment.outside.temperature","value":299.29999999999995} 
Jan 23 17:33:19 Front pattern: SFF 
Jan 23 17:33:19 Handle: {"path":"environment.outside.pressure","value":102181.7332549582} 
Jan 23 17:33:20 Handle: {"path":"navigation.position","value":{"longitude":-80.37339966666667,"latitude":27.663609833333332}} 
Jan 23 17:33:21 Handle: {"path":"navigation.gnss.antennaAltitude","value":-39.6} `

**Comment by fahrsoft** [comment]
Does the code do anything to check the ability to write to the folder, or to verify it exists? Is there somewhere else I should try writing to that might have better results? The note says "absolute or from the plugin directory," so if I were to make it from the plugin directory, where would I find that and what would an example path be?

**Comment by VladimirKalachikhin** [comment]
It's not about the path.
After `naivegpxlogger Subscribed to navigation.trip.logging` must be starting of logging by SignalK subscribing mechanism. But this is not happening. This may be the case if the path 'navigation.trip.logging' is missing at this moment. 
I set the delay to as much as 5 seconds between setting the path and subscribing. It didn't help.

Now again I don't know why the subscription doesn't work. Do you have the latest version of SignalK?

**Comment by fahrsoft** [comment]
All up to date. When I went into the SignalK Installer it did a bunch of post-installation updates. I've restarted everything since then, apt updates and upgrades, etc. It's all up to date and no warnings (other than OpenPlotter is not set to auto-start and the logs are big). Still, no change in the outputs for this. 
Is there a way to see a list of subscriptions or something like that? Where does 'navigation.trip.logging' come from?

**Comment by VladimirKalachikhin** [comment]
'navigation.trip.logging' is created by naivegpxlogger plugin.
No way to see all subscriptions but it's not necessary: `naivegpxlogger Subscribed to navigation.trip.logging` printed after subscribing - with no errors.

Try it to make sure that the subscription really doesn't work:
[index.js.TXT](https://github.com/VladimirKalachikhin/naiveGPXlogger/files/10491599/index.js.TXT)

**Comment by fahrsoft** [comment]
`
Jan 24 11:55:29 2023-01-24T16:55:29.093Z naivegpxlogger plugin stopped, logging= false 
Jan 24 11:55:29 2023-01-24T16:55:29.096Z naivegpxlogger plugin started, now logging is true log dir is /gpxlogs/ 
Jan 24 11:55:29 2023-01-24T16:55:29.099Z naivegpxlogger Do establish navigation.trip.logging to true 
Jan 24 11:55:29 POST /skServer/plugins/naivegpxlogger/config 200 5.351 ms - 45 
Jan 24 11:55:29 GET /skServer/plugins 304 12.712 ms - - 
Jan 24 11:55:29 2023-01-24T16:55:29.117Z naivegpxlogger Do establish navigation.trip.logging to false 
Jan 24 11:55:29 2023-01-24T16:55:29.118Z naivegpxlogger Subscribed to navigation.trip.logging 
Jan 24 11:55:32 GET /skServer/debugKeys 200 5.213 ms - - 
Jan 24 11:55:34 Handle: {"path":"navigation.gnss.antennaAltitude","value":22.6} 
Jan 24 11:55:41 Handle: {"path":"environment.outside.temperature","value":301.89} 
Jan 24 11:55:43 Front pattern: SFR 
Jan 24 11:55:43 Handle: {"path":"environment.outside.pressure","value":102473.63952723806} 
Jan 24 11:55:45 Handle: {"path":"navigation.position","value":{"longitude":-80.37326983333334,"latitude":27.663864666666665}} `

**Comment by VladimirKalachikhin** [comment]
Yes, no navigation.trip.logging events are coming up.
But event must up if navigation.trip.logging is present on start plugin, or if navigation.trip.logging is changed. One of these events necessarily occurs when the plugin starts.

I should consult the authors SignalK.

**Comment by VladimirKalachikhin** [comment]
I tested plugin in fresh VenusOS Large on real Raspberry Pi.
It works...

**Comment by tkurki** [comment]
Let's see about debugging this further. I am curious as to what is causing the `Front pattern: SFR` and the `Handle:` log messages. Any idea?

This is what I would try next
- disable all other plugins and restart. does it work?
- set the `Directory with tracks` to a directory that does not yet exist, like `tracks1` and restart. Is the directory created?
- what are the contents of your `$HOME/.signalk/plugin-config-data/naivegpxlogger.json`?
- is there `navigation.trip.logging` in Data Browser and what is its value?
- try sending `{"updates":[{"values":[{"path":"navigation.trip.logging","value":{"status":true}}]}]}` via Data Fiddler to the server. you can toggle between true and false - is the directory and log file created?

**Comment by fahrsoft** [comment]
Disable all other plugins and restart: No difference.
Set the directory to /tracks1/ and restart: error: False to create /tracks1/ by Permission Denied.
Set the directory to tracks1 and restart: created directory in $HOME/.signalk/node_modules/naivegpxlogger/ but no files created.
naivegpxlogger.json contents:
`{
  "configuration": {
    "logging": true,
    "trackFrequency": 0,
    "minmove": 0.3,
    "trackTimeout": 15,
    "metadata": {
      "skipperName": false
    },
    "depthProp": {
      "enable": false,
      "feature": "Depth below transducer (DBT)",
      "fixDepth": false
    },
    "trackDir": "tracks1",
    "everyDay": true
  },
  "enabled": true,
  "enableDebug": true,
  "enableLogging": true
}`
navigation.trip.logging in data browser shows: 
`{
  "status": false,
  "logFile": null
}`
Sending that in Data Fiddler did set it to true and create the gpx file. Changing to false, then changing the directory back to /gpxlogs did also start creating a file in that directory.

**Comment by VladimirKalachikhin** [comment]
The plugin sends: 
`{"context":"vessels.self","updates":[{"values":[{"path":"navigation.trip.logging","value":{"status":true,"logFile":null}}],"source":{"label":"naivegpxlogger"},"timestamp":"2023-02-06T09:44:57.053Z"}]}`

[index.js.TXT](https://github.com/VladimirKalachikhin/naiveGPXlogger/files/10661846/index.js.TXT)

**Comment by fahrsoft** [comment]
Same thing... it doesn't start a new file. Log: 
`
Feb 06 08:31:43 Handle: {"path":"navigation.position","value":{"longitude":-77.79516583333333,"latitude":26.929866}} 
Feb 06 08:31:43 Handle: {"path":"navigation.gnss.antennaAltitude","value":-2} 
Feb 06 08:31:44 2023-02-06T13:31:44.780Z naivegpxlogger plugin stopped, logging= false 
Feb 06 08:31:44 POST /skServer/plugins/naivegpxlogger/config 200 6.680 ms - 45 
Feb 06 08:31:44 2023-02-06T13:31:44.784Z naivegpxlogger [updSKpath] by app.handleMessage sended: {"context":"vessels.self","updates":[{"values":[{"path":"navigation.trip.logging","value":{"status":false,"logFile":null}}],"source":{"label":"naivegpxlogger"},"timestamp":"2023-02-06T13:31:44.784Z"}]} 
Feb 06 08:31:44 GET /skServer/plugins 200 12.201 ms - - 
Feb 06 08:31:47 2023-02-06T13:31:47.018Z naivegpxlogger plugin stopped, logging= false 
Feb 06 08:31:47 2023-02-06T13:31:47.019Z naivegpxlogger plugin started, now logging is true log dir is /gpxlogs/ 
Feb 06 08:31:47 2023-02-06T13:31:47.020Z naivegpxlogger [updSKpath] by app.handleMessage sended: {"context":"vessels.self","updates":[{"values":[{"path":"navigation.trip.logging","value":{"status":true,"logFile":null}}],"source":{"label":"naivegpxlogger"},"timestamp":"2023-02-06T13:31:47.020Z"}]} 
Feb 06 08:31:47 POST /skServer/plugins/naivegpxlogger/config 200 3.556 ms - 45 
Feb 06 08:31:47 2023-02-06T13:31:47.024Z naivegpxlogger [updSKpath] by app.handleMessage sended: {"context":"vessels.self","updates":[{"values":[{"path":"navigation.trip.logging","value":{"status":false,"logFile":null}}],"source":{"label":"naivegpxlogger"},"timestamp":"2023-02-06T13:31:47.024Z"}]} 
Feb 06 08:31:47 2023-02-06T13:31:47.026Z naivegpxlogger [doLogging] Subscribed to navigation.trip.logging via app.subscriptionmanager.subscribe by send {"context":"vessels.self","subscribe":[{"path":"navigation.trip.logging","format":"delta","policy":"instant","minPeriod":0}]} `

Sending the data fiddler command it starts logging (to the previous day's file). Restarting everything, same thing. If I send the data fiddler, then it started a new file.

**Comment by fahrsoft** [comment]
Also, and let me know if I should open another ticket, now that I have a GPX file (and I closed out the XML from the end tags not being on the file) I tried to open it in Garmin Basecamp and it says it's not a valid GPX file. Am I missing something? Is there some way that the file can always have the end tags in it and the appending clear those out, add the new lines, and put the end tags back in?

**Comment by VladimirKalachikhin** [comment]
Strange log... Ok, more debugging printing.
[index.js.TXT](https://github.com/VladimirKalachikhin/naiveGPXlogger/files/10679317/index.js.TXT)
Pls, switch "Write log" via Server -> Plugin Config interface and show the log.

**Comment by fahrsoft** [comment]
`
Feb 09 11:31:00 2023-02-09T16:31:00.239Z naivegpxlogger plugin stopped, logging= false 
Feb 09 11:31:00 2023-02-09T16:31:00.242Z naivegpxlogger plugin started, now logging is true log dir is /gpxlogs/ 
Feb 09 11:31:00 2023-02-09T16:31:00.243Z naivegpxlogger Initializing SignalK path 
Feb 09 11:31:00 2023-02-09T16:31:00.243Z naivegpxlogger [updSKpath] by app.handleMessage sended: {"context":"vessels.self","updates":[{"values":[{"path":"navigation.trip.logging","value":{"status":true,"logFile":null}}],"source":{"label":"naivegpxlogger"},"timestamp":"2023-02-09T16:31:00.243Z"}]} 
Feb 09 11:31:00 2023-02-09T16:31:00.245Z naivegpxlogger [doLogging] Subscribing to navigation.trip.logging via app.subscriptionmanager.subscribe by send {"context":"vessels.self","subscribe":[{"path":"navigation.trip.logging","format":"delta","policy":"instant","minPeriod":0}]} 
Feb 09 11:31:00 2023-02-09T16:31:00.245Z naivegpxlogger [doLogging] Subscribed to navigation.trip.logging 
Feb 09 11:31:00 POST /skServer/plugins/naivegpxlogger/config 200 2.790 ms - 45 
Feb 09 11:31:00 2023-02-09T16:31:00.248Z naivegpxlogger [updSKpath] by app.handleMessage sended: {"context":"vessels.self","updates":[{"values":[{"path":"navigation.trip.logging","value":{"status":false,"logFile":null}}],"source":{"label":"naivegpxlogger"},"timestamp":"2023-02-09T16:31:00.248Z"}]} 
Feb 09 11:31:00 2023-02-09T16:31:00.248Z naivegpxlogger [doOnControl] navigation.trip.logging event fired! 
Feb 09 11:31:00 2023-02-09T16:31:00.249Z naivegpxlogger [doOnControl] Need to turn off the recording, logging= false routeSaveName= null 
Feb 09 11:31:00 GET /skServer/plugins 200 12.041 ms - - 
Feb 09 11:31:06 Handle: {"path":"navigation.position","value":{"longitude":-77.36489366666666,"latitude":26.818334666666665}} 
Feb 09 11:31:13 GET /skServer/debugKeys 200 2.532 ms - - `

**Comment by VladimirKalachikhin** [comment]
For some reason this:
```
Feb 09 11:31:00 2023-02-09T16:31:00.243Z naivegpxlogger [updSKpath] by app.handleMessage sended:
{"context":"vessels.self","updates":[{"values":[{"path":"navigation.trip.logging","value":{"status":true,"logFile":null}}],"source":{"label":"naivegpxlogger"},"timestamp":"2023-02-09T16:31:00.243Z"}]}
```
passes without a trace.
Only [tkurki](https://github.com/tkurki) can give an idea.

**Comment by valkx** [comment]
Are there any new insights on this? I have the same issue and also have multiple sources for position.

**Comment by VladimirKalachikhin** [comment]
No. 
May be, the multiple sources is key? I'll try to simulate it.

**Comment by VladimirKalachikhin** [comment]
Work fine for me...
I create in SignalK two Data Connections with the same NMEA data. It works.
I create two Data Connections with different NMEA data. It works too.

**Comment by VladimirKalachikhin** [comment]
A somewhat strange question: what operating system are you using?

**Comment by fahrsoft** [comment]
Oh how I wish Linux would just do some simple stuff like have an “about” in the menus to answer such questions without getting out a keyboard. It’s 64-bit Debian, and it’s been recently updated, but I’m not sure beyond that. Tony MillerOn Mar 25, 2023, at 4:19 PM, VladimirKalachikhin ***@***.***> wrote:﻿
A somewhat strange question: what operating system are you using?

**Comment by VladimirKalachikhin** [comment]
Well, in Debian is a `tail` command. That's not the point..

**Comment by fahrsoft** [comment]
Debian 11 Bullseye 64-bit Tony MillerOn Mar 25, 2023, at 5:01 PM, VladimirKalachikhin ***@***.***> wrote:﻿
Well, in Debian is a tail command. That's not the point..

**Comment by VladimirKalachikhin** [comment]
If Debian -- may be AppArmor? SELinux?

**Comment by fahrsoft** [comment]
I’m not sure what any of that means. It’s the 64-bit OS from the raspberry Pi site. Tony MillerOn Mar 27, 2023, at 5:58 AM, VladimirKalachikhin ***@***.***> wrote:﻿
If Debian -- may be AppArmor? SELinux?

**Comment by VladimirKalachikhin** [comment]
Heh. Use these two links:
https://help.ubuntu.com/community/AppArmor#Disable_AppArmor_framework
https://wiki.debian.org/AppArmor/HowToUse#Disable_AppArmor

**Comment by fahrsoft** [comment]
Pretty sure AppArmor is not installed on Raspbian… I don’t find any commands for it.

**Comment by VladimirKalachikhin** [comment]
Ok, so that's no AppArmor.

**Comment by VladimirKalachikhin** [comment]
Fixed in 0.1.2
